### PR TITLE
ARROW-10393: [Rust] Apply fix for null reading in json reader for nested

### PR DESCRIPTION
**Pull request by vertexclique**
Apply fix for null reading in json reader for nested string builder variations.

### CODE REVIEW

File: rust/arrow/test/data/list_string_dict_nested_nulls.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{"machine": "a", "events": [null, "Elect Leader", "Do Ballot"]}
+{"machine": "b", "events": ["Do Ballot", null, "Send Data", "Elect Leader"]}
```
Comment (jorgecarleitao) [inline rust/arrow/test/data/list_string_dict_nested_nulls.json:2]: /r/oddlyspecific

**Review by jorgecarleitao — APPROVED** [review]
LGTM. Nice catch, @vertexclique

File: rust/arrow/test/data/list_string_dict_nested_nulls.json (hunk 1 above)
Comment (vertexclique) [reply rust/arrow/test/data/list_string_dict_nested_nulls.json:2]: lol, yes.

**Comment by jorgecarleitao** [comment]
@vertexclique , could you check clippy?

**Pull request closed without merging**